packages/devtools_app/test/flutter/timeline_screen_test.dart

```diff
@@ -41,62 +77,147 @@ void main() {
-          tester.state(find.byType(TimelineScreenBody));
+      await buildTimelineScreen(tester, TimelineMode.frameBased);
 
       final splitFinder = find.byType(Split);
 
       // Verify TimelineMode.frameBased content.
-      expect(state.controller.timelineMode, equals(TimelineMode.frameBased));
       expect(splitFinder, findsNothing);
       expect(find.text('Pause'), findsOneWidget);
       expect(find.text('Resume'), findsOneWidget);
       expect(find.text('Record'), findsNothing);
       expect(find.text('Stop'), findsNothing);
       expect(find.byType(FlutterFramesChart), findsOneWidget);
-      expect(find.byType(TimelineFlameChart), findsNothing);
+      expect(find.byKey(TimelineScreen.flameChartSectionKey), findsNothing);
       expect(find.byType(EventDetails), findsNothing);
 
       // Add a selected frame and ensure the flame chart and event details
       // section appear.
-      final mockData = MockFrameBasedTimelineData();
-      when(mockData.displayDepth).thenReturn(8);
-      when(mockData.selectedFrame).thenReturn(testFrame);
-      final controllerWithData = TimelineController()
-        ..frameBasedTimeline.data = mockData;
-      await tester.pumpWidget(wrapWithControllers(
-        TimelineScreenBody(),
-        timelineController: controllerWithData,
-      ));
+      await buildTimelineWithSelectedFrame(tester);
       expect(find.byType(FlutterFramesChart), findsOneWidget);
+      expect(find.byKey(TimelineScreen.flameChartSectionKey), findsOneWidget);
       expect(find.byType(TimelineFlameChart), findsOneWidget);
+      expect(find.byKey(TimelineScreen.recordingInstructionsKey), findsNothing);
       expect(find.byType(EventDetails), findsOneWidget);
 
       // Switch timeline mode and pump.
       await tester.tap(find.byType(Switch));
       await tester.pump();
 
       // Verify TimelineMode.full content.
-      expect(state.controller.timelineMode, equals(TimelineMode.full));
+      expect(
+        state.controller.timelineModeNotifier.value,
+        equals(TimelineMode.full),
+      );
       expect(find.text('Pause'), findsNothing);
       expect(find.text('Resume'), findsNothing);
       expect(find.text('Record'), findsOneWidget);
       expect(find.text('Stop'), findsOneWidget);
       expect(find.byType(FlutterFramesChart), findsNothing);
-      expect(find.byType(TimelineFlameChart), findsOneWidget);
+      expect(find.byKey(TimelineScreen.flameChartSectionKey), findsOneWidget);
+      expect(find.byType(TimelineFlameChart), findsNothing);
+      expect(
+        find.byKey(TimelineScreen.recordingInstructionsKey),
+        findsOneWidget,
+      );
       expect(find.byType(EventDetails), findsOneWidget);
 
       // Verify the state of the splitter.
       expect(splitFinder, findsOneWidget);
       final Split splitter = tester.widget(splitFinder);
       expect(splitter.initialFirstFraction, equals(0.6));
     });
+
+    testWidgets('record', (WidgetTester tester) async {
```

Comment:
What is required for this test to pass?  Explaining that in the test declaration can make tests much easier to understand at a glance.